src/Model/Linsen.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE FlexibleContexts, OverloadedStrings, LambdaCase  #-}
+
+module Model.Linsen
+  (
+    parse
+  , fetchAndCreateLinsen
+  )
+where
+
+import           Control.Monad                            ( (>=>)
+                                                          , (<=<)
+                                                          , zipWithM
+                                                          , ap
+                                                          )
+import           Control.Monad.Catch                      ( MonadThrow )
+import           Control.Monad.IO.Class                   ( MonadIO )
+import           Data.Aeson                               ( (.:)
+                                                          , withObject
+                                                          , Value
+                                                          )
+import           Data.Aeson.Types                         ( Parser
+                                                          , parseEither
+                                                          )
+import           Data.Bifunctor                           ( first )
+import qualified Data.ByteString.Lazy.Char8    as BL8
+import           Data.Functor                             ( (<&>) )
+import           Data.Text.Lazy                           ( Text
+                                                          , replace
+                                                          , strip )
+import           Data.Thyme.Calendar                      ( Day )
+import           Network.HTTP.Req
+import           Model.Types                              ( NoMenu(..)
+                                                          , Menu(..)
+                                                          , Restaurant
+                                                            ( Restaurant
+                                                            )
+                                                          )
+import           Util                                     ( menusToEitherNoLunch )
+import           Data.Thyme.Calendar.WeekDate             ( toWeekDate )
+
+fetch
+  :: (MonadHttp m, MonadIO m, MonadThrow m)
+  => m Value  -- ^ A JSON response or horrible crash
+fetch =
+  req
+    GET
+    (https "cafe-linsen.se" /: "api" /: "menu")
+    NoReqBody
+    jsonResponse
+    mempty
+  <&> responseBody
+
+parse
+  :: Day                  -- ^ Day to parse
+  -> Value                -- ^ JSON result from `fetch`
+  -> Either NoMenu [Menu] -- ^ Either list of parsed `Menu`s or `NoMenu` error
+parse day =
+    failWithNoMenu
+      (parseEither
+        (   withObject "Parse meals"
+        $   (.: "docs")
+        >=> (pure . (!! (6 - (\(_,_,a) -> a) (toWeekDate day))))
```

Comment:
Indices and arithmetic makes parsing so much better. Do we get the whole week in the json blob and then pick the correct day by index?

There's a way to get the third element from something by using the lens `_3`, but it's very much a case of... if you love lenses you can give it a go. :smile_cat:

Reply:
The whole week is in the blob.

I changed a bit of the code here, but haven't figured out how to index into the list with lenses without also introducing `Maybe`. Maybe clearly handling that case would be better for this code